src/d3a/models/market/market_structures.py

```diff
@@ -232,15 +224,15 @@ class TradeBidOfferInfo:
     propagated_offer_rate: float
     trade_rate: float
 
-    def to_json_string(self):
+    def to_json_string(self) -> str:
         return json.dumps(asdict(self), default=my_converter)
 
     @classmethod
-    def len(cls):
+    def len(cls) -> int:
         return cls.len()
 
 
-def trade_bid_info_from_json_string(info_string):
+def trade_bid_info_from_json_string(info_string) -> TradeBidOfferInfo:
```

Comment:
Similar comment as above: I would see this as a class method for `TradeBidOfferInfo`:
```python
trade_bid_offer = TradeBidOfferInfo.from_json(json_str)
```

Reply:
Nit: On another side: why is this called `TradeBidOfferInfo` and not `TradeInfo`? Do you have any idea about the reason?